ONSdigital/rm-collection-exercise-service
# Ras 829 supplementary data service pubsub consumer

# What and why?

This introduces a consumer for the Supplementary Dataset message from the Supplementary Data Service. 

# How to test?

- deploy the PR image to a sandbox environment (inc. the config and Terraform on the attached card) and ensure Spring Boot registers the subscription

```
Application is listening for Supplementary Data Service requests supplementary-data-service-<ENV>"
```

- send a message to the topic (see attached card for detail)

- ensure the message is consumed and mapped to the DTO

```
Mapping to Supplementary Dataset object successful SupplementaryDatasetDTO(...)
```

# Jira
https://jira.ons.gov.uk/browse/RAS-829
